charmdir.go

```diff
@@ -70,24 +70,24 @@ func ReadCharmDir(path string) (dir *CharmDir, err error) {
 	dir = &CharmDir{Path: path}
 	file, err := os.Open(dir.join("metadata.yaml"))
 	if err != nil {
-		return nil, err
+		return nil, errors.Annotatef(err, `issue reading "metadata.yaml" file`)
```

Comment:
I prefer these annotations to always be only what we were doing at the time. We'll know there was an issue/error based on the log level we choose for it, making "error doing x" or "issue with y" redundant.

So it would be `reading "metadata.yaml" file`